src/Sylius/Bundle/ProductBundle/Form/DataTransformer/ProductsToProductAssociationsTransformer.php

```diff
@@ -18,60 +18,65 @@
 use Sylius\Component\Product\Model\ProductAssociationInterface;
 use Sylius\Component\Product\Model\ProductAssociationTypeInterface;
 use Sylius\Component\Product\Model\ProductInterface;
-use Sylius\Component\Product\Repository\ProductRepositoryInterface;
 use Sylius\Component\Resource\Factory\FactoryInterface;
 use Sylius\Component\Resource\Repository\RepositoryInterface;
 use Symfony\Component\Form\DataTransformerInterface;
 use Webmozart\Assert\Assert;
 
+/**
+ * @implements DataTransformerInterface<Collection<array-key, ProductAssociationInterface>, array<string, Collection<array-key, ProductInterface>>>
+ */
 final class ProductsToProductAssociationsTransformer implements DataTransformerInterface
 {
-    /** @var Collection<array-key, ProductAssociationInterface> */
+    /** @var Collection<array-key, ProductAssociationInterface>|null */
     private ?Collection $productAssociations = null;
 
+    /**
+     * @param FactoryInterface<ProductAssociationInterface> $productAssociationFactory
+     * @param RepositoryInterface<ProductAssociationTypeInterface> $productAssociationTypeRepository
+     */
     public function __construct(
-        private FactoryInterface $productAssociationFactory,
-        private ProductRepositoryInterface $productRepository,
-        private RepositoryInterface $productAssociationTypeRepository,
+        private readonly FactoryInterface $productAssociationFactory,
+        private readonly RepositoryInterface $productAssociationTypeRepository,
     ) {
```

Comment:
You have removed one of the parameters, is it a violation of BC?

Reply:
If so, you should add the appropriate deprecation to 1.14